File: dataset/codenet/Java/Code/s125837071.java
import java.util.Scanner;

public class Main {

	@SuppressWarnings("resource")
	public static void main(String args[]) {
		Scanner scanner = new Scanner(System.in);
		int n = scanner.nextInt();
		long ans = 1;
		int mod = (int) Math.pow(10, 9) + 7;
		for (int i = 2; i <= n; i++) {
			for (int j = 2; (i % j != 0 && j < i) || j == i; j++) {
				if (j == i) {
					int s = 0;
					for (int k = 1; k <= n; k++) {
						s += findS(k, i);

					}

					ans = (ans * (s + 1) ) % mod;

				}
			}
		}
		System.out.println(ans);
	}

	private static int findS(int k, int i) {
		int s = 0;
		for (int j = k; j % i == 0; j /= i) {

			s++;

		}
		return s;
	}
}

File: output/deepseek_chat/codenet/direct_translation/java/python/successful_translation/s125837071.py

import math

def main():
    scanner = input()
    n = int(scanner)
    ans = 1
    mod = int(math.pow(10, 9)) + 7
    for i in range(2, n + 1):
        j = 2
        while (i % j != 0 and j < i) or j == i:
            if j == i:
                s = 0
                for k in range(1, n + 1):
                    s += findS(k, i)
                ans = (ans * (s + 1)) % mod
            j += 1
    print(ans)

def findS(k, i):
    s = 0
    while k % i == 0:
        s += 1
        k //= i
    return s

if __name__ == "__main__":
    main()

